lib/spark_api/models/listing.rb

```diff
@@ -235,6 +235,10 @@ def respond_to?(method_symbol, include_all=false)
         end
       end
 
+      def batch_photo_delete(photoIds, args={})
```

Comment:
I don't think it quite fits with Mr. Black's definition, but using the bang seems consistent with the existing usage in the project, which tried to mimic the same patterns in active record for `save!` and validation errors:

https://github.com/sparkapi/spark_api/blob/master/lib/spark_api/models/listing.rb#L143-L178

ActiveRecord seems to also have a [destroy!](http://apidock.com/rails/ActiveRecord/Persistence/destroy!), which seems similar in this instance. We don't have much for validation here, but the API does enforce some rules based on MLS photo minimums, and conditions for deleting the primary photo.

The bang variant is missing from the [existing delete method](https://github.com/sparkapi/spark_api/blob/master/lib/spark_api/models/photo.rb#L53-L55), but I propose it's just as relevant there. One distinction between the singular vs bulk delete is that we cannot "undo" the bulk operation which makes it more dangerous, but I'm not positive we make that part of the criteria.